scenarios/kubernetes_build.py

```diff
@@ -30,8 +31,49 @@ def check(*cmd):
     print >>sys.stderr, 'Run:', cmd
     subprocess.check_call(cmd)
 
+def check_output(*cmd):
+    """Log and run the command, raising on errors, return output"""
+    print >>sys.stderr, 'Run:', cmd
+    return subprocess.check_output(cmd)
+
+def check_build_exists(gcs, suffix):
+    """ check if a k8s/federation build with same version
+        already exists in remote path
+    """
+    if not os.path.exists('hack/print-workspace-status.sh'):
+        print >>sys.stderr, 'hack/print-workspace-status.sh not found, continue'
+        return False
+
+    version = ''
+    try:
+        match = re.search(
+            r'gitVersion ([^\n]+)',
+            check_output('hack/print-workspace-status.sh')
+        )
+        if match:
+            version = match.group(1)
+    except subprocess.CalledProcessError as exc:
+        # fallback with doing a real build
+        print >>sys.stderr, 'Failed to get k8s version, continue: %s' % exc
+        return False
+
+    if version:
+        if not gcs:
+            gcs = 'gs://kubernetes-release-dev'
+        mode = 'ci'
+        if suffix:
+            mode = os.path.join(mode, suffix)
+        gcs = os.path.join(gcs, mode, version)
+        try:
+            check('gsutil', 'ls', gcs)
```

Comment:
I wonder if we should suppress stderr. I foresee
```
CommandException: One or more URLs matched no objects.
```
potentially confusing users.

Reply:
do we have a standard way to suppress stderr?

Reply:
something like https://stackoverflow.com/a/11269627, probably?

Reply:
yeah, that works

```
../test-infra/scenarios/kubernetes_build.py --suffix=boo
Run: ('hack/print-workspace-status.sh',)
Run: ('gsutil', 'ls', 'gs://kubernetes-release-dev/ci/boo/v1.4.0-alpha.2.26172+054490609f57c5')
gcs path gs://kubernetes-release-dev/ci/boo/v1.4.0-alpha.2.26172+054490609f57c5 does not exist yet, continue
Run: ('make', 'clean')
...
```